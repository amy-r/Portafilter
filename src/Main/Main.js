import React from 'react';
import { key } from '../Utilities/.api-key'
import { body, text } from '../Utilities/emailBody'

export const Main = (props) => {
  console.log(key)
  const sgMail = require('@sendgrid/mail');
  sgMail.setApiKey(key);
  console.log(sgMail)
  const msg = {
    to: 'jpquinn605@gmail.com',
    from: 'hello@amethystcoffee.co',
    subject: 'Hello, Jordan!',
    text: text,
    html: body
  };
  sgMail.send(msg);

  return (
    <div>
      <h1>
        Main Container
      </h1>
    </div>
  )
}

