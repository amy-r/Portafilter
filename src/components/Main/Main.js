import React, { Component } from 'react';
import { key } from '../../Utilities/api-key';
import { body, text } from '../../Utilities/emailBody';
import { Route, Switch } from 'react-router-dom';
import CardContainer from '../CardContainer/CardContainer';
import Form from '../Form/Form';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import Login from '../Login/Login';
import CoffeeForm from '../CoffeeForm/CoffeeForm';

const sgMail = require('@sendgrid/mail');
  sgMail.setApiKey(key);
  const msg = {
    to: 'jpquinn605@gmail.com',
    from: 'hello@amethystcoffee.co',
    subject: 'Hello, Jordan!',
    text: text,
    html: body
  };
  sgMail.send(msg);


export class Main extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div>
        <Switch>
          <Route path='/form' component={Form} />
          <Route path='/coffee-form' component={CoffeeForm} />
          <Route path='/current-roasters' component={CardContainer} />
          <Route path='/' component={Login} />
        </Switch>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  user: state.user
})

export default withRouter(connect(mapStateToProps, null)(Main))

