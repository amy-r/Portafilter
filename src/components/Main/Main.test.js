import React from 'react';
import ReactDOM from 'react-dom';
import { Main, mapStateToProps, mapDispatchToProps } from './Main';
import { shallow, mount } from 'enzyme';

describe('Main', () => {
  it('should render correctly', () => {

    const mockUser = {
      userName: 'Jordan',
      userEmail: 'Jordan@Project.com',
      userPhoto: 'photo-url',
      userId: 'fancy encryption'
    }

    const wrapper = shallow(<Main user={mockUser}/>)
    expect(wrapper).toMatchSnapshot();
  })
})

describe('MSTP', () => {
  it('should define user props for the container', () => {
    const userName= "Sharon Jones";
    const userEmail= "sharon@gmail.com";
    const mockStore = {user: [ {name, userEmail} ]};
    const expected = [ {name, userEmail} ];
    const mapped = mapStateToProps(mockStore);
    expect(mapped.user).toEqual(expected)
  })
})