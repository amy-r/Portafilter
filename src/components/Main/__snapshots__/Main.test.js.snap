// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    user={
        Object {
            "userEmail": "Jordan@Project.com",
            "userId": "fancy encryption",
            "userName": "Jordan",
            "userPhoto": "photo-url",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component) />,
        <Switch>
          <Route
                    component={[Function]}
                    path="/form"
          />
          <Route
                    component={[Function]}
                    path="/coffee-form"
          />
          <Route
                    component={[Function]}
                    path="/current-roasters"
          />
          <Route
                    component={[Function]}
                    path="/"
          />
</Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/form"
/>,
            <Route
              component={[Function]}
              path="/coffee-form"
/>,
            <Route
              component={[Function]}
              path="/current-roasters"
/>,
            <Route
              component={[Function]}
              path="/"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/form",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/coffee-form",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/current-roasters",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component) />,
          <Switch>
            <Route
                        component={[Function]}
                        path="/form"
            />
            <Route
                        component={[Function]}
                        path="/coffee-form"
            />
            <Route
                        component={[Function]}
                        path="/current-roasters"
            />
            <Route
                        component={[Function]}
                        path="/"
            />
</Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/form"
/>,
              <Route
                component={[Function]}
                path="/coffee-form"
/>,
              <Route
                component={[Function]}
                path="/current-roasters"
/>,
              <Route
                component={[Function]}
                path="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/form",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/coffee-form",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/current-roasters",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
